### PR TITLE
Rewrite typescript declarations

### PR DESCRIPTION
**Pull request by david-driscoll**
This is a rewrite of the .d.ts files from the ground up, using the existing src files as a reference.  Most things have ported over, others have been updated to match RxJS 3.0.

There a are a few things left to do still, so this PR is very much a work in progress.  I'd love to hear thoughts to see if you guys think this is a right direction or not.

- [x] Automate creation of .d.ts files (core, all, aggregates, etc) as part of the grunt build
- [x] Support ES6 use cases (iterables, map, set, etc)
- [x] Support Node.js use cases (streams)

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@david-driscoll__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft Open Technologies, Inc. and real humans are currently evaluating your PR.</p>

TTYL, MSOTBOT;

File: .settings/tasks.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (david-driscoll) [inline .settings/tasks.json]: I plan to drop this file and squash the commits before the PR is finished.

**Comment by david-driscoll** [comment]
I've tested this against the Omnisharp-atom code base (we use RxJs quiet a bit) everything looks to be working pretty well from a definitions point of view.

**Comment by mattpodwysocki** [comment]
@david-driscoll currently working on this branch if you are curious about the changes https://github.com/Reactive-Extensions/RxJS/tree/scheduler_refactor

**Comment by david-driscoll** [comment]
Cool, that looks fairly straightforward.

**Comment by mattpodwysocki** [comment]
@david-driscoll can you please fix the build and the nuget packages please?

**Comment by david-driscoll** [comment]
sure thing.

**Comment by electricessence** [comment]
Port of Linq.js to typescript with generics:
https://github.com/electricessence/TypeScript.NET/tree/master/source/System.Linq

**Pull request merged**